src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigPaths.cs

```diff
@@ -92,7 +92,31 @@ private ClientConfigPaths(string exePath, bool includeUserConfig)
                 }
             }
 
-            if (!string.IsNullOrEmpty(ApplicationUri))
+            string externalConfigPath = AppDomain.CurrentDomain.GetData("APP_CONFIG_FILE") as string;
+            if (!string.IsNullOrEmpty(externalConfigPath))
+            {
+                if (Uri.IsWellFormedUriString(externalConfigPath, UriKind.Absolute))
+                {
+                    Uri externalConfigUri = new Uri(externalConfigPath);
+                    if (externalConfigUri.IsFile)
+                    {
+                        ApplicationConfigUri = externalConfigUri.LocalPath;
+                    }
+                }
+                else
+                {
+                    // Uri constructor would throw for relative paths but full framework doesn't.
+                    // We will mimic full framework behavior here.
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && externalConfigPath.Length >= 2 && externalConfigPath.StartsWith('\\') && externalConfigPath[1] != '\\')
```

Comment:
Is there some way to avoid the OS check here? We don't normally do those. What if on Unix it started with '/' ... should it trim whatever the Path.DirectorySeparatorChar is?

Reply:
Just always trim any amount of leading `Path.DirectorySeparatorChar` and you should be good.

Reply:
I think for Unix we will never go into this code anyway since that would mean absolute path which should be covered by the first check. I'll go ahead and trim all separators on the left - that I think this should cover 99% of the cases